etc/make_small_lib.pl

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+use File::Basename 'basename';
+use File::Glob 'bsd_glob';
+use File::Spec;
+
+my $sep = File::Spec->catfile('', '');
+my %fast_functions = (
+       s_mp_balance_mul            => 1,
+       s_mp_exptmod_fast           => 1,
+       s_mp_invmod_fast            => 1,
+       s_mp_karatsuba_mul          => 1,
+       s_mp_karatsuba_sqr          => 1,
+       s_mp_montgomery_reduce_fast => 1,
+       s_mp_mul_digs_fast          => 1,
+       s_mp_mul_high_digs_fast     => 1,
+       s_mp_sqr_fast               => 1,
+       s_mp_toom_mul               => 1,
+       s_mp_toom_sqr               => 1
+   );
+my @dependency_list = ();
+
+sub uniq {
+    my %seen;
+    grep !$seen{$_}++, @_;
+}
+
+sub read_file {
+  my $f = shift;
+  open my $fh, "<", $f or die "FATAL: read_rawfile() cannot open file '$f': $!";
+  binmode $fh;
+  return do { local $/; <$fh> };
+}
```

Comment:
File::Slurper?

Reply:
To quote the aptly named Cenk K. Uygur of TyT fame: "Have at it, Hoss!" ;-)

This file contains a lot of C&P from `helper.pl` which has been glued together with nothing more but vague memories from bygone times, so don't expect too much, thank you.